src/main/java/org/broadinstitute/dsm/model/BSPKit.java

```diff
@@ -0,0 +1,157 @@
+import org.broadinstitute.dsm.util.SystemUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class BSPKit {
+    private static Logger logger = LoggerFactory.getLogger(BSPKit.class);
+
+    public Optional<BSPKitStatus> getKitStatus(@NonNull String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Checking label " + kitLabel);
+        BSPKitQueryResultDao bspKitQueryResultDao = new BSPKitQueryResultDao();
+        Optional<BSPKitQueryResultDto> bspKitQueryResult = bspKitQueryResultDao.getBSPKitQueryResult(kitLabel);
+        Optional<BSPKitStatus> result = Optional.empty();
+        if (bspKitQueryResult.isEmpty()) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+        }
+        else {
+            BSPKitQueryResultDto maybeBspKitQueryResult = bspKitQueryResult.get();
+            if (StringUtils.isNotBlank(maybeBspKitQueryResult.getParticipantExitId())) {
+                String message = "Kit of exited participant " + maybeBspKitQueryResult.getBspParticipantId() + " was received by GP.<br>";
+                notificationUtil.sentNotification(maybeBspKitQueryResult.getNotificationRecipient(), message, NotificationUtil.DSM_SUBJECT);
+                result = Optional.of(new BSPKitStatus(BSPKitStatus.EXITED));
+            }
+            else if (StringUtils.isNotBlank(maybeBspKitQueryResult.getDeactivationDate())) {
+                result = Optional.of(new BSPKitStatus(BSPKitStatus.DEACTIVATED));
+            }
+        }
+        return result;
+    }
+
+    public boolean canReceiveKit(@NonNull String kitLabel) {
+        BSPKitQueryResultDao bspKitQueryResultDao = new BSPKitQueryResultDao();
+        Optional<BSPKitQueryResultDto> bspKitQueryResult = bspKitQueryResultDao.getBSPKitQueryResult(kitLabel);
+        if (bspKitQueryResult.isEmpty()) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+            return false;
+        }
+        else {
+            BSPKitQueryResultDto maybeBspKitQueryResult = bspKitQueryResult.get();
+            if (StringUtils.isNotBlank(maybeBspKitQueryResult.getParticipantExitId()) || StringUtils.isNotBlank(maybeBspKitQueryResult.getDeactivationDate())) {
+                logger.info("Kit can not be received.");
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Optional<BSPKitInfo> receiveBSPKit(String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Trying to receive kit " + kitLabel);
+        BSPKitQueryResultDao bspKitQueryResultDao = new BSPKitQueryResultDao();
+        Optional<BSPKitQueryResultDto> bspKitQueryResult = bspKitQueryResultDao.getBSPKitQueryResult(kitLabel);
+        bspKitQueryResult.map(maybeBspKitQueryResult -> {
+            if (StringUtils.isNotBlank(maybeBspKitQueryResult.getDdpParticipantId())) {
```

Comment:
using guard condition instead of putting huge logic in `if` block is better for visibility:
`if (StringUtils.isBlank(maybeBspKitQueryResult.getDdpParticipantId())) throw new RuntimeException()
//logic goes here
`

Reply:
I have to say I like the "guard condition" =)